### DISCUSSION TITLE
Hiring a Mastodon developer to make a custom server for scientific research

### DISCUSSION THREAD

**Discussion by aaronbergen** [mastodon/mastodon#31204] [General]
Hi,

I want to deploy a custom Mastodon server.  I'm a C++ software developer, but completely new to Mastodon and Ruby and React.  So I'd like to hire an expert who knows the code and how to make a fork that will allow this customization to be applied as a patch to the Mastodon code so we stay current.  The custom code will be FOSS so if any of it is useful for a general purpose Mastodon server it can be merged into the main branch.  But since it's fairly niche, I doubt that will happen, so want to maintain compatibility as much as possible with a patch.  I've attached the specifications.  Anybody interested can email me at aaron@factmission.org

Thanks.
[Mastadon Specs.pdf](https://github.com/user-attachments/files/16420321/Mastadon.Specs.pdf)

**Comment by mjankowski** [comment]
I scanned your thing. Thoughts...

- What's the rationale for wanting to run this on top of a mastodon instance vs its own custom citation/voting type thing?
- This is well well short of your doc, but FYI people can *Add a link to* their Orcid ID via their profile now, which maybe is better than nothing
- I suspect that doing this would be straightforward ... but doing this in a way that was resilient to staying current/upgraded to latest mastodon release would actually be somewhat maddening and not straightforward.
- Your question is 5 months old - are you still doing this, or is it dropped?
